public/views/partials/lib/commands/user/create.liquid

```diff
@@ -11,34 +11,43 @@
 {% liquid
   function object = 'modules/user/lib/commands/user/create/build', email: email, password: password, hook_params: hook_params
   function object = 'modules/user/lib/commands/user/create/check', object: object
+  function results = 'modules/core/lib/commands/hook/fire', hook: 'user_create_validate', params: hook_params
```

Comment:
The idea is to setup a "pre_user_create" hook called `user_create_validate`. It's job is to validate all input needed for user creation without performing any real action or permanent modification.
If it finds an error, we go back to the user and present the errors. Otherwise call the real command and continue as normal (where input will be checked again, not ideal, not terrible).

Reply:
Nice!